### PR TITLE
Rework CI/CD to one file and add deploy job

### PR DESCRIPTION
**Pull request by ks129**
This is generally not a good idea to have jobs that depend on each other on different workflow files, so I merged them into one file. Also added deployment job to Kubernetes, so we can deploy automatically! I also gave duckyfied names to jobs.

### CODE REVIEW

File: .github/workflows/duckinuous-quackration.yml [hunk 1]
```diff
@@ -86,3 +85,76 @@ jobs:
...
+        with:
+          context: .
+          file: ./Dockerfile
+          push: true
+          cache-from: type=registry,ref=ghcr.io/python-discord/quackstack:main
+          tags: |
+            ghcr.io/python-discord/quackstack:main
+            ghcr.io/python-discord/quackstack:${{ steps.sha_tag.outputs.tag }}
+
+  deploy:
```
Comment (HassanAbouelela) [inline .github/workflows/duckinuous-quackration.yml:126]: We should add something here to limit this to master, not PRs, and possibly use an environment.

**Review by HassanAbouelela — CHANGES_REQUESTED** [review]
I appreciate the jokes, but wouldn't that be confusing?

File: .github/workflows/duckinuous-quackration.yml (hunk 1 above)
Comment (HassanAbouelela) [reply .github/workflows/duckinuous-quackration.yml:126]: Nevermind, missed it in the build. Still the point about environments.

Comment (ks129) [reply .github/workflows/duckinuous-quackration.yml:126]: This has already the `needs: [build]` and `build` job defines constraint, so if build gets skipped, then this skips too.

**Review by HassanAbouelela — APPROVED** [review]

**Review by Akarys42 — CHANGES_REQUESTED** [review]
I agree with Scal, I'd like at least the workflow file names to stay readable. I don't mind the action name though.

**Review by Akarys42 — APPROVED** [review]
LGTM

**Pull request merged**